### ISSUE TITLE
naic gxgx

### ISSUE THREAD

**Issue by 404289680** [JDAI-CV/fast-reid#367]
恭喜大佬，我也拿的你的代码solo的，不过就拿了十几名，能请教下这比赛核心问题在哪么，怎么解决的，是你用的那个超参数搜索策略么

**Comment by L1aoXingyu** [comment]
这两天会开源决赛方案，可以关注一下

**Comment by 404289680** [comment]
请问代码会更新么

**Comment by L1aoXingyu** [comment]
codebase 基本用的就是现在这一套

**Comment by 404289680** [comment]
竟然没有额外的trick，那是我失误了，估计是我数据处理没搞好，期待早日开源！

**Comment by L1aoXingyu** [comment]
唯一新加的东西就是 pairwise cosface

**Comment by 404289680** [comment]
这个会比circle softmax好很多吗

**Comment by L1aoXingyu** [comment]
circle softmax + pairwise cosface 一起用

**Comment by 404289680** [comment]
代码里看到了cosface，这个东西要怎么一起用，会介绍么，还是说直接同时用就行，没有需要注意的点

**Comment by 404289680** [comment]
直接在embedding head，接两个classifier么

**Comment by L1aoXingyu** [comment]
pairwise cosface 是类似 triplet loss 的 metric loss，不需要接 fc

**Comment by 404289680** [comment]
请问有paper么，还是说是你比赛期间想出的idea，现在的代码有办法用上么

**Comment by 404289680** [comment]
感觉是无监督那边借鉴的？

**Comment by L1aoXingyu** [comment]
就是 circle loss 那篇 paper 里面，一个 uniform loss function，可以推出来 cosface 的分类loss版本，也可以推出来 pairwise 版本

**Comment by 404289680** [comment]
好的，谢谢，我去看看

**Comment by L1aoXingyu** [comment]
部分代码已经更新了

**Comment by 404289680** [comment]
wow，感谢感谢，我看看

**Comment by 404289680** [comment]
能拿第一硬实力确实强，期待早日看到方案分享！！！

**Comment by 404289680** [comment]
能额外请教个问题么，我复赛的时候有几点很疑惑，不知道是我弄错了还是确实不work，一个是fliptest，一个是开源数据集，你可以work么

**Comment by L1aoXingyu** [comment]
flip test 可以提升0.3-0.5，直接加公开数据集提交1个点左右，做个变换再加进去可以稍微涨多一点，不过也不明显。

**Comment by 404289680** [comment]
好的，请问是A榜的提升么

**Comment by 404289680** [comment]
好的，谢谢，看来没想想中提升的多
